main.go

```diff
@@ -106,6 +107,25 @@ type powerDNSProviderConfig struct {
 	//
 	// +optional
 	Timeout int `json:"timeout"`
+
+	// AllowedZones is the list of zones that may be edited. If the list is
+	// empty, all zones are permitted.
+	AllowedZones []string `json:"allowed-zones"`
+}
+
+// IsAllowedZone checks if the webhook is allowed to edit the given zone, per
+// AllowedZones setting.
+func (cfg powerDNSProviderConfig) IsAllowedZone(zone string) bool {
+	if len(cfg.AllowedZones) == 0 {
+		return true
+	}
+
+	for _, allowed := range cfg.AllowedZones {
+		if strings.HasSuffix(zone, allowed) {
```

Comment:
I would propose improving the logic on this to something like:

```suggestion
		if zone == allowed || strings.HasSuffix(zone, fmt.Sprintf(".%s", allowed))
```

Otherwise, the suffix match will result in `notexample.org.` matching `example.org.`.

Reply:
Good catch! Fix and tests added.